### PR TITLE
fix($theme-default): Fix slots syntax

### LINKED ISSUE #1963: Carbon ads are not displayed anymore
- [ ] I confirm that this is an issue rather than a question.

## Bug report

Carbon ads are not displayed anymore on the vuepress docs. 

#### Other relevant information

- Output of `npx vuepress info` in my VuePress project:

### PR DESCRIPTION
**Pull request by kefranabg**
**Summary**

Fix #1963 

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [ ] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: packages/@vuepress/theme-default/layouts/Layout.vue [hunk 1]
```diff
@@ -21,11 +21,11 @@
     >
       <slot
         name="sidebar-top"
-        #top
+        slot="top"
       />
```
Comment (ludanxer) [inline packages/@vuepress/theme-default/layouts/Layout.vue]: The new `v-slot` syntax can only be added to a `<template>`, so the code below should fix it
```
<template #top>
  <slot name="sidebar-top"/>
</template>
```

File: packages/@vuepress/theme-vue/layouts/Layout.vue [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <template>
   <ParentLayout>
-    <CarbonAds #sidebar-top/>
-    <BuySellAds #page-bottom/>
+    <CarbonAds slot="sidebar-top"/>
+    <BuySellAds slot="page-bottom"/>
```
Comment (ludanxer) [inline packages/@vuepress/theme-vue/layouts/Layout.vue]: We can only use the new `v-slot` on the component when name is _default_
```
<template #sidebar-top>
  <CarbonAds/>
</template>
```

**Comment by kefranabg** [comment]
Hi @ludanxer, thanks for doing a review 🙏
Actually, I'm wondering if the new slot syntax is really more readable in this specific case 🤔

**Comment by ludanxer** [comment]
Hi @kefranabg ,

I just noticed that PR #1951 already tried to fix this with the new syntax.

File: packages/@vuepress/core/lib/node/App.js [hunk 3]
```diff
@@ -154,7 +154,7 @@ module.exports = class App {
       .use(require('./internal-plugins/frontmatterBlock'))
       .use('container', {
         type: 'slot',
-        before: info => `<template #${info}>`,
+        before: info => `<template slot="${info}">`,
```
[the remark this answers is not in the document]
Comment (ulivz) [reply packages/@vuepress/core/lib/node/App.js]: It's right here, here we're providing slots instead of consuming slots.

**Comment by ulivz** [comment]
Just closed this in favor of #1950, and this pull request would try to fix 2 different regression issues introduced by https://github.com/vuejs/vuepress/commit/c17c70e96af5542edba86032c22cc67348251d78

**Pull request closed without merging**